### PR TITLE
e2e: move funs of framework/statefulset to e2e/apps & e2e/upgrades

### PR DESCRIPTION
**Pull request by clarklee92**
Signed-off-by: clarklee92 <clarklee1992@hotmail.com>

**What type of PR is this?**
 /kind cleanup

**What this PR does / why we need it**:
e2e: move funs of framework/statefulset to e2e/apps & e2e/upgrades
**Which issue(s) this PR fixes**:

Ref #86052

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/e2e/apps/statefulset.go [hunk 1]
```diff
@@ -52,8 +53,27 @@ const (
 
 	// Timeout for reads from databases running on stateful pods.
 	readTimeout = 60 * time.Second
+
+	// StatefulSetPoll is a poll interval for StatefulSet tests
+	StatefulSetPoll = 10 * time.Second
```
Comment (SataQiu) [inline test/e2e/apps/statefulset.go]: Thanks @clarklee92 
Small nits. Others lgtm.

Do these variables have to be exported？
If it is not referenced by other packages, we can set it to private.

File: test/e2e/apps/statefulset.go [hunk 2]
```diff
@@ -52,8 +53,27 @@ const (
...
+	// StatefulSetTimeout is a timeout interval for StatefulSet operations
+	StatefulSetTimeout = 10 * time.Minute
```
Comment (SataQiu) [inline test/e2e/apps/statefulset.go]: ditto

File: test/e2e/apps/statefulset.go [hunk 3]
```diff
@@ -52,8 +53,27 @@ const (
...
+	// StatefulPodTimeout is a timeout for stateful pods to change state
+	StatefulPodTimeout = 5 * time.Minute
```
Comment (SataQiu) [inline test/e2e/apps/statefulset.go]: ditto

File: test/e2e/apps/wait.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2esset "k8s.io/kubernetes/test/e2e/framework/statefulset"
+)
+
+// WaitForPartitionedRollingUpdate waits for all Pods in set to exist and have the correct revision. set must have
+// a RollingUpdateStatefulSetStrategyType with a non-nil RollingUpdate and Partition. All Pods with ordinals less
+// than or equal to the Partition are expected to be at set's current revision. All other Pods are expected to be
+// at its update revision.
+func WaitForPartitionedRollingUpdate(c clientset.Interface, set *appsv1.StatefulSet) (*appsv1.StatefulSet, *v1.PodList) {
```
Comment (SataQiu) [inline test/e2e/apps/wait.go]: Do these funs have to be exported？
If it is not referenced by other packages, we can set it to private.

File: test/e2e/apps/wait.go [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+			}
+		}
+		return true, nil
+	})
+	return set, pods
+}
+
+// WaitForStatus waits for the StatefulSetStatus's ObservedGeneration to be greater than or equal to set's Generation.
+// The returned StatefulSet contains such a StatefulSetStatus
+func WaitForStatus(c clientset.Interface, set *appsv1.StatefulSet) *appsv1.StatefulSet {
```
Comment (SataQiu) [inline test/e2e/apps/wait.go]: ditto

File: test/e2e/apps/wait.go [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+			set = set2
+			return true, nil
+		}
+		return false, nil
+	})
+	return set
+}
+
+// WaitForPodNotReady waits for the Pod named podName in set to exist and to not have a Ready condition.
+func WaitForPodNotReady(c clientset.Interface, set *appsv1.StatefulSet, podName string) (*appsv1.StatefulSet, *v1.PodList) {
```
Comment (SataQiu) [inline test/e2e/apps/wait.go]: ditto

File: test/e2e/apps/wait.go [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+			}
+		}
+		return false, nil
+	})
+	return set, pods
+}
+
+// WaitForRollingUpdate waits for all Pods in set to exist and have the correct revision and for the RollingUpdate to
+// complete. set must have a RollingUpdateStatefulSetStrategyType.
+func WaitForRollingUpdate(c clientset.Interface, set *appsv1.StatefulSet) (*appsv1.StatefulSet, *v1.PodList) {
```
Comment (SataQiu) [inline test/e2e/apps/wait.go]: ditto

File: test/e2e/apps/wait.go [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+			}
+			return false, nil
+		}
+		return true, nil
+	})
+	return set, pods
+}
+
+// WaitForRunningAndNotReady waits for numStatefulPods in ss to be Running and not Ready.
+func WaitForRunningAndNotReady(c clientset.Interface, numStatefulPods int32, ss *appsv1.StatefulSet) {
```
Comment (SataQiu) [inline test/e2e/apps/wait.go]: ditto

Comment (clarklee92) [reply test/e2e/apps/wait.go]: Updated, but I thought it might be used externally in the future.

**Comment by SataQiu** [comment]
/lgtm
/assign @oomichi
PTAL :)

File: test/e2e/upgrades/apps/statefulset.go [hunk 9]
```diff
@@ -18,6 +18,7 @@ package upgrades
 
 import (
 	"github.com/onsi/ginkgo"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (oomichi) [inline test/e2e/upgrades/apps/statefulset.go]: This should be moved to line 24 as the same import group.

File: test/e2e/apps/wait.go [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apps
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
```
Comment (oomichi) [inline test/e2e/apps/wait.go]: nit: need a blank line between imports of other repos and imports of k/k repo.

Comment (clarklee92) [reply test/e2e/apps/wait.go]: Yep, I am going to turn down `go importer` in my IDE 🤣.
Thank you for your careful review.

**Review by oomichi — COMMENTED** [review]
Thanks for updating,

/lgtm
/approve

**Pull request merged**